hathor/transaction/base_transaction.py

```diff
@@ -521,6 +521,7 @@ def validate_basic(self, skip_block_weight_verification: bool = False) -> bool:
 
         If no exception is raised, the ValidationState will end up as `BASIC` and return `True`.
         """
+        self._update_height_metadata()
```

Comment:
I think the caller should make sure the metadata is correctly filled. In this case, we might add some asserts here.

Reply:
Outdated conversation. This code has already been removed.